components/ws-manager/pkg/manager/monitor.go

```diff
@@ -987,52 +1024,47 @@ func (m *Monitor) finalizeWorkspaceContent(ctx context.Context, wso *workspaceOb
 				volumeSnapshotTime = time.Now()
 			}
 			if createdVolumeSnapshot {
-				backoff := wait.Backoff{
-					Steps:    30,
-					Duration: 100 * time.Millisecond,
-					Factor:   1.5,
-					Jitter:   0.1,
-					Cap:      10 * time.Minute,
+				waitTimeout := 10 * time.Minute
+				ctx, cancel := context.WithTimeout(context.Background(), waitTimeout)
+				defer cancel()
+
+				if m.notifyPod[wso.Pod.Name] == nil {
+					m.notifyPod[wso.Pod.Name] = make(chan string)
 				}
-				log = log.WithField("VolumeSnapshot.Name", pvcVolumeSnapshotName)
-				err = wait.ExponentialBackoff(backoff, func() (bool, error) {
-					var volumeSnapshot volumesnapshotv1.VolumeSnapshot
-					err := m.manager.Clientset.Get(ctx, types.NamespacedName{Namespace: m.manager.Config.Namespace, Name: pvcVolumeSnapshotName}, &volumeSnapshot)
-					if err != nil {
-						if k8serr.IsNotFound(err) {
-							// volumesnapshot doesn't exist yet, retry again
-							return false, nil
+
+				for {
+					select {
+					case pvcVolumeSnapshotContentName = <-m.notifyPod[wso.Pod.Name]:
+						readyVolumeSnapshot = true
+						break
```

Comment:
this `break` is not needed here, since it automatically will break out of `select` here